optaplanner-core/src/main/java/org/optaplanner/core/impl/score/buildin/hardsoftdouble/HardSoftDoubleScoreDefinition.java

```diff
@@ -20,11 +20,16 @@
 
 import org.optaplanner.core.api.score.buildin.hardsoftdouble.HardSoftDoubleScore;
 import org.optaplanner.core.api.score.buildin.hardsoftdouble.HardSoftDoubleScoreHolder;
+import org.optaplanner.core.api.score.holder.ScoreHolder;
 import org.optaplanner.core.config.score.trend.InitializingScoreTrendLevel;
 import org.optaplanner.core.impl.score.definition.AbstractScoreDefinition;
 import org.optaplanner.core.impl.score.inliner.ScoreInliner;
 import org.optaplanner.core.impl.score.trend.InitializingScoreTrend;
 
+/**
+ * @deprecated Double-based scores are deprecated as floating point numbers can't represent a decimal number correctly.
```

Comment:
The real reason they are deprecated is because "floating point numbers are not associative (A + B + C == C + B + A) and therefore they are not compatible with incremental score calculation".